### PR TITLE
Fix logging avro serde for struct columns 

### PR DESCRIPTION
**Pull request by hzding621**
## Summary

When multiple struct columns exist in the same avro schema (e.g. same join) that contain sub-columns with the same name, a suffix REPEATED_{i} is added. This PR fixes an issue in this scenario where the suffix is not handled, causing logging and log-flattener job failures. 

## Why / Goal

Fix logging and log-flattener job failures. 

## Test Plan

- [x] Added Unit Tests
- [x] Covered by existing CI
- [x] Integration tested

## Reviewers

@airbnb/airbnb-chronon-maintainers

### CODE REVIEW

File: spark/src/test/scala/ai/chronon/spark/test/SchemaEvolutionTest.scala [hunk 1]
```diff
@@ -171,6 +171,48 @@ class SchemaEvolutionTest extends TestCase {
...
   }
 
+  def createStructGroupBy(namespace: String, spark: SparkSession, name: String): GroupByTestSuite = {
+    val schema = StructType(
+      "struct_schema",
+      Array(
+        StructField("id", LongType),
+        StructField(name, ListType(StructType(name = "s", fields = Array(StructField("i", LongType))))),
+        StructField("ds", StringType)
+      )
```
Comment (pengyu-hou) [inline spark/src/test/scala/ai/chronon/spark/test/SchemaEvolutionTest.scala:190]: Can we add another nested struct test case with potential duplicate names like below? 
```
{ event_type,
 user_id, 
context: {user_id, timestamp}
}
```

**Review by pengyu-hou — APPROVED** [review]
Thanks for the fix. Left one comment for unit test.

File: spark/src/test/scala/ai/chronon/spark/test/SchemaEvolutionTest.scala (hunk 1 above)
Comment (hzding621) [reply spark/src/test/scala/ai/chronon/spark/test/SchemaEvolutionTest.scala:190]: Great idea
